optimum/habana/transformers/generation/stopping_criteria.py

```diff
@@ -37,26 +44,50 @@ def gaudi_MaxLengthCriteria_call(
                 f"maximum length ({self.max_position_embeddings}). Depending on the model, you may observe "
                 "exceptions, performance degradation, or nothing at all."
             )
-    is_done = 1 if is_done else 0
-    return torch.full((input_ids.shape[0],), is_done, device=input_ids.device, dtype=torch.int8)
+        return create_return_const_tensor(input_ids, is_done)
 
 
 def gaudi_MaxNewTokensCriteria_call(
     self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs
-) -> torch.BoolTensor:
+) -> Union[torch.BoolTensor, bool]:
     token_idx = kwargs.get("token_idx", None)
     if token_idx is not None:
-        is_done = token_idx >= self.max_length
+        assert not kwargs["out_type_tensor"]
+        return token_idx >= self.max_length
     else:
         is_done = input_ids.shape[-1] >= self.max_length
-    is_done = 1 if is_done else 0
-    return torch.full((input_ids.shape[0],), is_done, device=input_ids.device, dtype=torch.int8)
+        return create_return_const_tensor(input_ids, is_done)
+
+
+def gaudi_MaxTimeCriteria_call(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> Union[torch.BoolTensor, bool]:
+    is_done = time.time() - self.initial_timestamp > self.max_time
+    if kwargs["out_type_tensor"]:
+        return create_return_const_tensor(input_ids, is_done)
+    else:
+        return is_done
+    
+def gaudi_EosTokenCriteria_call(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> Union[torch.BoolTensor, bool]:
+    self.eos_token_id = self.eos_token_id.to(input_ids.device)
+    is_done = torch.isin(input_ids[:, -1], self.eos_token_id)
+    if kwargs["out_type_tensor"]:
+        return is_done.byte()
+    else:
+        return torch.all(is_done).item()
 
+def needs_tensor_output(token_idx, ignore_eos, eos_token_id) -> bool:
+    if token_idx is None:
+        return not ignore_eos and eos_token_id is not None
+    else:
+        # token_idx is present, so we have static shapes, so using single boolean
+        False
 
 def gaudi_StoppingCriteriaList_call(
     self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs
-) -> torch.BoolTensor:
-    is_done = torch.full((input_ids.shape[0],), 0, device=input_ids.device, dtype=torch.int8)
+) -> Union[torch.BoolTensor, bool]:
+    kwargs["out_type_tensor"] = needs_tensor_output(kwargs.get("token_idx",None), kwargs.get("ignore_eos",True), kwargs.get("eos_token_id",None))
```

Comment:
```suggestion
    kwargs["needs_tensor_output"] = needs_tensor_output(kwargs.get("token_idx",None), kwargs.get("ignore_eos",True), kwargs.get("eos_token_id",None))
```
Let's simply call this kwarg the same